channeldb/kvdb/etcd/bucket.go

```diff
@@ -0,0 +1,29 @@
+package etcd
+
+import (
+	"bytes"
+)
+
+func isValidKey(key []byte) bool {
+	return !bytes.ContainsAny(key, "/~")
```

Comment:
does this mean we need to encode binary keys or values so they don't mess with the control characters? seems like we need to pick an encoding scheme that doesn't collide with `{'b': 98, 'v': 118, '$': 36, '/': 47, '~': 126}`

Reply:
We might do that, or simply hash part of the keys eg. v/path/key => value_prefix + hash(path) + key. This will change key locality though (not sure if this is a problem)

Reply:
Other possibly viable idea is to use a lenght/value encoding for path parts and the key.

Reply:
@Roasbeef just pointed out there's a LevelDB implementation for walletdb, where the approach is pretty nice and flat: https://github.com/btcsuite/btcd/blob/06e5c43499a2f30a1f7d7c3b49de0073dfbbdd1f/database/ffldb/db.go#L565

I'll adopt something like this instead.

Reply:
So binary keys are supported for some time now.